test/compilationTests/zeppelin/crowdsale/Crowdsale.sol

```diff
@@ -92,7 +92,7 @@ contract Crowdsale {
   }
 
   // @return true if the transaction can buy tokens
-  function validPurchase() internal view returns (bool) {
+  function validPurchase() internal returns (bool) {
     uint256 current = block.number;
     bool withinPeriod = current >= startBlock && current <= endBlock;
     bool nonZeroPurchase = msg.value != 0;
```

Comment:
This is non-view because of `msg.value` IIRC, but why doesn't it require `payable` then?

This might be something we need to think about a bit.